### PR TITLE
fix(actor-critic): emit the policy that produced each recorded action

### LINKED ISSUE #382: run_actor_critic_from_arrays returns policies[t] evaluated at next_observations[t] after the update, not the distribution that produced actions[t]
## Summary

`run_actor_critic_from_arrays` (`alberta_framework/core/actor_critic.py`) documents `ActorCriticArrayResult.policies` as "Per-step policy probabilities" alongside `actions` ("Per-step actions") and `values` ("Per-step previous-state value estimates"). `actions[t]` is sampled at `observations[t]` and `values[t]` is the value at `observations[t]`, but the scan emits `result.policy` — `ActorCriticUpdateResult.policy`, documented as the policy "at the new observation" — i.e. the **post-update** softmax at `next_observations[t]`. The three per-step arrays are not on the same time index. The sibling continuous runner already records `current_mean, current_sigma` at `obs`, aligned with `actions[t]`.

Any per-step policy diagnostic computed from the returned arrays — `log π(a_t|s_t)`, policy entropy at the visited state, KL between arms, importance ratios for off-policy correction — reads the wrong row.

## Falsifier (deterministic, pristine `main` @ `4f75cc8`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.actor_critic import ActorCriticAgent, ActorCriticConfig, run_actor_critic_from_arrays
agent = ActorCriticAgent(ActorCriticConfig(n_actions=3, actor_step_size=0.5, gamma=0.9))
state = agent.init(feature_dim=2, key=jr.key(5)).replace(
    actor_weights=jnp.array([[3.0, 0.0], [0.0, 3.0], [0.0, 0.0]]))
obs = jnp.array([[1., 0.], [0., 1.], [1., 0.]]); nxt = jnp.array([[0., 1.], [1., 0.], [0., 1.]])
r = run_actor_critic_from_arrays(agent, state, obs, jnp.ones(3), jnp.array([False]*3), nxt)
print("actions:", r.actions)
print("policies[0]:", r.policies[0], " policy at observations[0]:", agent.policy(state, obs[0]))
```

Output on `main`:

```
actions: [0 1 0]
policies[0]: [0.0452785 0.909443  0.0452785]   policy at observations[0]: [0.909443  0.0452785 0.0452785]
```

`policies[0]` puts 0.045 on the action that was actually taken with probability 0.909; `log π(a_t|s_t)` from the returned arrays is `[-3.09, -0.72, -3.09]` against a ground truth of `[-0.09, -0.09, -0.72]` — wrong by 3 nats on two of three steps, with a sign that flips with the trajectory.

Expected: `policies[t]` is the pre-update policy at `observations[t]` (the distribution that produced `actions[t]`), as the continuous runner already does; on a rejected update it is neutralized like `actions[t]`.

## Dedup

`gh issue/pr list --search "run_actor_critic_from_arrays" / "policies"`: nothing on this surface (adjacent AC PRs #202, #64 are the 0×inf lane). No open PR touches `core/actor_critic.py`.

Fix PR follows (failing-test-first).

### PR DESCRIPTION
**Pull request by Prabhat1308**
Closes #382.

## Issue

`run_actor_critic_from_arrays` emitted `result.policy` — the post-update softmax at `next_observations[t]` — as `policies[t]`, so the policy row was off by one state and one update relative to `actions[t]` and `values[t]` (`policies[0] = [0.045, 0.909, 0.045]` for an action taken with probability 0.909; `log π(a_t|s_t)` wrong by 3 nats).

## New state

The scan captures the pre-update agent policy at `observations[t]` — from `agent.start` when sampling, from `agent.policy(started_state, obs)` on the fixed-actions branch — and emits it as `policies[t]`, mirroring the continuous runner's `current_mean/current_sigma`. On a rejected update the row is neutralized to zeros, exactly as `actions[t]` is neutralized to `0`. The docstrings (`b23b417`, maintainer) state that `policies[t]` is the sampling distribution only when this runner samples the actions; with fixed behavior actions it is the agent's own policy at that state, not provenance for the external behavior policy. `values`, `td_errors`, `actions`, and the carried state are unchanged.

Failing-test-first: `test_run_actor_critic_from_arrays_sampled_policies_align_with_actions` (replays the scan step-by-step and asserts `policies[t] == agent.policy(state_t, observations[t])`) fails on `main` and passes here; `test_run_actor_critic_from_arrays_fixed_actions_matches_loop` now also pins the fixed-actions rows to the pre-update agent policy.

## Evidence

Falsifier from #382 at this head:

```text
reported == at observations[t] ? True   reported == at next_obs[t] ? False
log pi(a_t|s_t) from returned arrays: [-0.0949 -0.0949 -0.7177]
log pi(a_t|s_t) ground truth        : [-0.0949 -0.0949 -0.7177]
```

Verification at `b23b4178ea5b9661a930df96ffebd002b5a3f111`:

```text
.venv/bin/python -m pytest tests/test_actor_critic.py tests/test_continuous_actor_critic.py -q -o addopts=""   -> 28 passed
.venv/bin/python -m ruff check .                                                                            -> All checks passed!
.venv/bin/python -m mypy                                                                                    -> Success: no issues found in 164 source files
```

`core/actor_critic.py` is imported by the robot track and stays importable (signature unchanged); it is not a registered evidence source; no benchmark lane, seeds, or `outputs/` artifacts were touched; no `CHANGELOG.md` edit (maintainer-recorded).

CLI sessions cannot mint GitHub attachment URLs, so the run output above is inline rather than attached.

- [x] logs: inline above (focused pytest, ruff, mypy, falsifier output at the evidence head)

Provider/model/client: anthropic / claude-fable-5 / Claude Code 2.1.233 (contribute-to-asi skill revision 72e4239e).

### CODE REVIEW

**Review by lalalune — CHANGES_REQUESTED** [review]
One blocking API-contract correction at exact head b6f697a. The new ActorCriticArrayResult doc says policies[t] is "the distribution that produced actions[t]" for every call, but run_actor_critic_from_arrays explicitly supports fixed actions from pre-collected behavior logs. In that branch this PR computes the current agent target policy at observations[t]; it cannot know the external behavior distribution that produced the supplied action. The code's state/time alignment is correct, but the new public provenance claim is false and could lead downstream importance/log-probability calculations to treat a target policy as behavior data.

Please document policies as the pre-update agent policy aligned to observations[t], adding that it is the sampling distribution only when actions=None, and add a fixed-actions regression pinning policies[t] to agent.policy(state_t, observations[t]) without calling it behavior. The on-policy alignment test is sound.

**Review by lalalune — APPROVED** [review]
Approved at exact repaired head b23b417. The scan arithmetic is unchanged from the alignment fix; the public contract now correctly distinguishes on-policy sampled actions from externally supplied/pre-collected actions, and the fixed-action regression proves returned rows are current-agent evaluations rather than unknown behavior-policy provenance. Exact-head focused actor-critic/continuous matrix: 27 passed; targeted Ruff and strict mypy passed. Fresh CI is authorized and queued.

**Comment by Prabhat1308** [comment]
Agreed on the provenance wording — your `b23b417` on this branch is exactly the correction I would have made (docstrings now say the rows are the pre-update agent policy at `observations[t]`, the sampling distribution only when this runner samples the actions, and the fixed-actions loop test pins the rows to `agent.policy(state_t, observations[t])`). I have adopted that head as-is (my duplicate local commit was dropped) and updated the body to evidence head `b23b417`. 28 passed across the two actor-critic suites, ruff/mypy clean. Re-review requested.

**Review by lalalune — APPROVED** [review]
Re-review at exact head b23b417: READY. The result and runner docs now correctly distinguish the pre-update agent policy from behavior provenance when fixed actions come from external logs, and the fixed-action regression pins that target-policy alignment with a deliberately unlikely supplied action. The original on-policy state/update alignment remains sound. Exact refreshed head: 27 actor/continuous-actor tests passed; prior combined adjacent panel was 244 passed; Ruff and strict mypy passed, with clean current-main composition. No registered evidence source or output artifact changes.
